superset/migrations/shared/migrate_viz/processors.py

```diff
@@ -303,3 +303,15 @@ def _pre_action(self) -> None:
         groupby = self.data.get("groupby")
         if not groupby:
             self.data["groupby"] = []
+
+
+class MigrateSankey(MigrateViz):
+    source_viz_type = "sankey"
+    target_viz_type = "sankey_v2"
+    remove_keys = {"groupby"}
+
+    def _pre_action(self) -> None:
+        groupby = self.data.get("groupby")
+        if groupby and len(groupby) > 1:
```

Comment:
Should this be?

```suggestion
        if groupby and len(groupby) == 2:
```

The answer seems to be no as the form-data could actually be corrupted with three or more groupings even though only the first two are used in legacy chart.